### PR TITLE
Follow up test fixes (dimensions as ndarray inputs)/misc fixes

### PR DESCRIPTION
**Pull request by agibsonccc**
Java imports clean up, dimensions related fixes for resolving dimensions based on multiple inputs

Adds generateFakeData(..) to DynamicCustomOp to accommodate ops that need different kinds of fake data as placeholders to ensure proper op execution.

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

Running TF import tests 

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [X] Relevant tests for your changes are passing.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml [hunk 1]
```diff
@@ -204,6 +204,9 @@
                     <classPaths>
                         <classPath>${project.build.outputDirectory}</classPath>
                     </classPaths>
+            <!--        <compilerOptions>
+                        <compilerOption>-g</compilerOption>
+                    </compilerOptions>-->
```
Comment (treo) [inline nd4j/nd4j-backends/nd4j-backend-impls/nd4j-native/pom.xml]: I guess you forgot to remove this?

File: platform-tests/src/test/java/org/nd4j/imports/tfgraphs/TFGraphTestAllHelper.java [hunk 2]
```diff
@@ -151,20 +155,30 @@ protected static void checkOnlyOutput(Map<String, INDArray> inputs, Map<String,
...
+/*
+        if(modelName.equals("math_mul_order")) {
+            GraphDef g  = GraphDef.parseFrom(IOUtils.toByteArray(oldModel.toURI()));
+            oldForComparison = TFGraphMapper.importGraph(oldModel);
+            oldOutputs = oldForComparison.outputAll(inputs);
+            TensorflowIRGraph tensorflowIRGraph = new TensorflowIRGraph(g,tensorflowFrameworkImporter.getOpDefList(),tensorflowFrameworkImporter.getRegistry());
+            TensorflowIRGraphRunner tensorflowIRGraphRunner = new TensorflowIRGraphRunner(tensorflowIRGraph,new ArrayList<>(inputs.keySet()),new ArrayList<>(oldOutputs.keySet()));
+            //outputs = tensorflowIRGraphRunner.run(inputs);
+            System.out.println();
+        }
```
Comment (treo) [inline platform-tests/src/test/java/org/nd4j/imports/tfgraphs/TFGraphTestAllHelper.java]: Lots of commented out code here and an empty println. 

This doesn't look merge ready.

**Review by treo — CHANGES_REQUESTED** [review]
A few things here:

1. You've got another test result file in here: `libnd4j/tests_cpu/surefire-reports/TEST-cpu-results.xml`
2. Why are you using fully qualified class names for standard library like String in the IR files? 
3. It looks like you've also kept a few more tempFiles from tests around in the platform-tests directory

**Comment by agibsonccc** [comment]
@treo ok got most of the nits there. Regarding the IR classes, those are auto generated. I'm not sure if you can configure the protobuf generator to not do that. I'll check though.

**Review by treo — APPROVED** [review]

**Pull request merged**